oauth-common/src/main/java/io/strimzi/kafka/oauth/validator/JWTSignatureValidator.java

```diff
@@ -302,7 +355,9 @@ private void fetchKeys() {
                 cache = newCache;
             }
             lastFetchTime = System.currentTimeMillis();
+
         } catch (Throwable ex) {
+            addJwksHttpMetricErrorTime(ex, requestStartTime);
```

Comment:
Catching a `Throwable` and rethrowing it as `RuntimeException` is a bit weird. I'm not familiar with this code but is it really what we want to do?

Reply:
This method is called as part of ScheduledExecutorService scheduled task which I don't want to stop working, which happens if the task method throws any exception. Additionally, I want the stacktrace to include the error message with some context info - the message set on the thrown RuntimeException.

I guess your proposal would be to do:
```
        } catch (Throwable ex) {
            addJwksHttpMetricErrorTime(ex, requestStartTime);
            if (ex instanceof Error) {
               throw (Error) ex;
            }
            throw new RuntimeException("Failed to fetch public keys needed to validate JWT signatures: " + keysUri, ex);
        }
```

But then there is `setupExecutorAndFetchInitialKeys()` method that rethrows the caught exception. There is some ergonomics in play when only RuntimeExceptions are possible that they don't have to be declared, so it just makes for a cleaner code IMHO to wrap anything here into RuntimeException as at the end of the day all is handled the same by my code higher up the stack.